python/taichi/examples/graph/mpm88_graph.py

```diff
@@ -113,12 +113,12 @@ def init_particles(x: ti.any_arr(field_dim=1), v: ti.any_arr(field_dim=1),
     if not args.baseline:
         print('running in graph mode')
         # Build graph
-        sym_x = ti.graph.Arg(ti.graph.ArgKind.NDARRAY, 'x', 'f32', element_shape=(2, ))
-        sym_v = ti.graph.Arg(ti.graph.ArgKind.NDARRAY, 'v', 'f32', element_shape=(2, ))
-        sym_C = ti.graph.Arg(ti.graph.ArgKind.NDARRAY, 'C', 'f32', element_shape=(2, 2))
-        sym_J = ti.graph.Arg(ti.graph.ArgKind.NDARRAY, 'J', 'f32', element_shape=())
-        sym_grid_v = ti.graph.Arg(ti.graph.ArgKind.NDARRAY, 'grid_v', 'f32', element_shape=(2, ))
-        sym_grid_m = ti.graph.Arg(ti.graph.ArgKind.NDARRAY, 'grid_m', 'f32', element_shape=())
+        sym_x = ti.graph.Arg(ti.graph.ArgKind.NDARRAY, 'x', ti.f32, element_shape=(2, ))
```

Comment:
How about having `ScalarArg` and `NdarrayArg` in the future? That way we don't have to have a generic `element_shape`

Reply:
SG! Btw we have similar structs at https://github.com/taichi-dev/taichi/blob/master/taichi/aot/module_data.h#L60-L83 but they were tweaked for launch kernels individually instead of graph execution. Will send out a followup PR to separate `ScalarArg` and `NdarrayArg` for the graph use case!